Extensions/IISWebAppDeploy/Src/Tasks/SqlDacpacDeploy/DeployToSqlServer.ps1

```diff
@@ -81,22 +75,88 @@ function GetScriptToRun
 
     if ($taskType -eq "dacpac")
     {
-        $invokeMain = "Execute-DacpacDeployment -dacpacFile `"$dacpacFile`" -targetMethod $targetMethod -serverName `"$serverName`" -databaseName `"$databaseName`" -authscheme $authscheme -sqlServerCredentials `$sqlServerCredentials -connectionString `"$connectionString`" -publishProfile `"$publishProfile`" -additionalArguments `"$additionalArguments`""
-
+        
         $sqlPackageScript = Get-Content TaskModuleSqlUtility\SqlPackageOnTargetMachines.ps1 | Out-String
 
+        try{
+            $sqlSplatArguments = @{
+                dacpacFile=$dacpacFile 
+                targetMethod=$targetMethod 
+                serverName=$serverName
+                databaseName=$databaseName
+                authscheme=$authscheme 
+                sqlServerCredentials="`$sqlServerCredentials"
+                connectionString=$connectionString 
+                publishProfile=$publishProfile 
+                additionalArguments=$additionalArguments
+            }
+            $jsonSqlSplatArguments = $sqlSplatArguments | ConvertTo-Json
+            $remoteArgsInit = "`$remoteJsonSqlSplattedArgs = '$jsonSqlSplatArguments'
```

Comment:
Can you explain why is this required? Did you test the task with additional arguments and for all the different input scenarios?